scripts/maint/practracker/practracker.py

```diff
@@ -131,8 +131,10 @@ def main():
     # 3) Go through all the files and report problems if they are not exceptions
     found_new_issues = consider_all_metrics(files_list)
 
+    # If new issues were found, try to give out some advice to the developer on how to resolve it.
     if (found_new_issues):
-        print("practracker FAILED")
+        new_issues_str = "practracker FAILED as indicated by the problem lines above. Please use the exceptions file ({}) to find any previous state of these problems. If you are unable to fix the underlying best-practices issue right now then you need to either update the relevant exception line or add a new one.".format(exceptions_file)
```

Comment:
With your permission, I'll tweak this string a little post-merge.  The way it is written now, it suggests that the preferred solution to a bad practice is to add an exception, rather than to fix the problem.  I think there should be a new section in doc/HACKING about this, with suggestions for how to fix each kind of problem.

Reply:
Yes, no problem with tweaking that string.